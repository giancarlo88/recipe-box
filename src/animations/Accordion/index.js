import React from 'react'
import { Motion, spring } from 'react-motion'

const Accordion = (Component, initialHeight) =>
  class AccordionComponent extends React.Component {
    constructor(props) {
      super(props)
      this.height = initialHeight
    }

    componentDidMount() {
      this.height = this.element ? this.element.scrollHeight : 0
    }

    render() {
      const { isSelected } = this.props
      return (
        <Motion
          defaultStyle={{ x: this.height }}
          style={{ x: isSelected ? spring(this.height) : spring(95) }}
        >
          {(interpolatingStyle) => (
            <div
              ref={(elem) => {
                this.element = elem
              }}
              style={{ height: interpolatingStyle.x, overflow: 'hidden' }}
            >
              <Component {...this.props}/>
            </div>
          )}
        </Motion>
      )
    }
  }

export default Accordion
