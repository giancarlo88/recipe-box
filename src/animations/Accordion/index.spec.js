import React from 'react'
import { create } from 'react-test-renderer'
import { shallow, mount } from 'enzyme'

import Accordion from '.'

function createNodeMock(element) {
  console.log(element)
  if (element.type === 'div') {
    return {
      scrollHeight: 100
    }
  }
}

describe('<Accordion /> HOC', () => {
  const initialHeight = 100
  class MockComponent extends React.Component {
    render() {
      return <div className='mock' ref={(elem) => (this.element = elem)}>Mock</div>
    }
  }

  it('should match snapshot', () => {
    const WrappedComponent = Accordion(MockComponent)
    const wrapper = mount(<WrappedComponent />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass the initialHeight down to the rendered child', () => {
    const WrappedComponent = Accordion(MockComponent, 100)
    const wrapper = mount(<WrappedComponent />)
    expect(wrapper.childAt(0).height).toBe(initialHeight)
  })
})
