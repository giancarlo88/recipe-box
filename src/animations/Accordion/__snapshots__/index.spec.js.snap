// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> HOC should match snapshot 1`] = `
<AccordionComponent>
  <Motion
    defaultStyle={
      Object {
        "x": undefined,
      }
    }
    style={
      Object {
        "x": Object {
          "damping": 26,
          "precision": 0.01,
          "stiffness": 170,
          "val": 95,
        },
      }
    }
  >
    <div
      style={
        Object {
          "height": undefined,
          "overflow": "hidden",
        }
      }
    >
      <MockComponent>
        <div
          className="mock"
        >
          Mock
        </div>
      </MockComponent>
    </div>
  </Motion>
</AccordionComponent>
`;
